packages/backend/src/config/app.ts

```diff
@@ -137,3 +138,21 @@ function parseRedisTlsConfig(
 
   return Object.keys(options).length > 0 ? options : undefined
 }
+
+function parseOrProvisionKey(keyPath: string): crypto.KeyObject {
+  if (keyPath !== '') {
+    try {
+      const key = crypto.createPrivateKey(readFileSync(keyPath))
+      const jwk = key.export({ format: 'jwk' })
+      if (jwk.crv === 'Ed25519') {
+        return key
+      } else {
+        console.log('Private key is not EdDSA-Ed25519 key. Generating new key.')
+      }
+    } catch {
+      console.log('Private key could not be loaded. Generating new key.')
+    }
+  }
+  const keypair = crypto.generateKeyPairSync('ed25519')
```

Comment:
Actually...
should this be storing the generated key to at least a tmp file?

Reply:
I was wondering something similar and wanted to make a comment but then had to run. 
I was wondering whether we should store the keys somewhere when Rafiki "goes down" for whatever reason. Is that what you want to use the tmp file for?
Also, where would you like to put the tmp file?

Reply:
I don't have a location in mind, but maybe whatever we choose could be the default value for `PRIVATE_KEY_PATH`